src/main/java/org/openstreetmap/atlas/checks/validation/relations/RouteRelationCheck.java

```diff
@@ -0,0 +1,845 @@
+            final String stopOrPlatform)
+    {
+        final List<AtlasEntity> allSigns = rel.members().stream()
+                .filter(member -> member.getRole().equals(stopOrPlatform))
+                .map(RelationMember::getEntity).collect(Collectors.toList());
+
+        final List<PolyLine> allEdgePolyLines = this.polylineRouteRel(rel).getAllPolyLines();
+        final List<Location> signLocationsFlagged = new ArrayList<>();
+        final List<AtlasEntity> allSignsEntitiesFlagged = new ArrayList<>();
+
+        for (final AtlasEntity entity : allSigns)
+        {
+            final List<Location> signLocations = new ArrayList<>();
+            if (entity instanceof AtlasItem)
+            {
+                final AtlasItem signPositions = (AtlasItem) entity;
+                signPositions.getRawGeometry().forEach(signLocations::add);
+            }
+
+            if (this.stopPlatformTooFarFromTrackCheck(signLocations, allEdgePolyLines))
+            {
+                allSignsEntitiesFlagged.add(entity);
+            }
+
+        }
+
+        return new TestStructureData(null, null, allSignsEntitiesFlagged, signLocationsFlagged,
+                null);
+    }
+
+    /**
+     * This is the helper function for checkStopPlatformTooFarFromTrack that do the check.
+     *
+     * @param signLocations
+     *            the relation entity supplied by the Atlas-Checks framework for evaluation
+     * @param allEdgePolyLines
+     *            indicate whether we want locations for stops or platforms
+     * @return a boolean indicating whether the stop or platform is too far from the track
+     */
+
+    private boolean stopPlatformTooFarFromTrackCheck(final List<Location> signLocations,
+            final List<PolyLine> allEdgePolyLines)
+    {
+        final Distance threshHold = Distance.meters(15.0);
+        SnappedLocation minSnap = null;
+        // Location minLoc = null
+
+        for (final Location location : signLocations)
+        {
+            for (final PolyLine edges : allEdgePolyLines)
+            {
+                final SnappedLocation snappedTo = location.snapTo(edges);
+                if (minSnap == null || snappedTo.compareTo(minSnap) < 0)
+                {
+                    minSnap = snappedTo;
+                }
+            }
+        }
+
+        return minSnap != null && minSnap.getDistance().isGreaterThan(threshHold);
+    }
+
+    /**
+     * @param routeRelation
+     *            the public transport route relation. this method checks whether the route relation
+     *            is contained in a route master
+     * @return an instance of CheckRouteMasterValues containing information about whether this
+     *         public transport route is contained in a route master
+     */
+    private boolean testContainedInRouteMasters(final Relation routeRelation)
+    {
+        final Iterable<Relation> relationsInAtlas = routeRelation.getAtlas().relations();
```

Comment:
Instead of going trough all the relation in the atlas you can just get the relations the route is a member of by calling `routeRelation.relations()`.